### PR TITLE
Update var-and-list.js

### PR DESCRIPTION
**Pull request by qxsck**
add some blocks, fix format

### CODE REVIEW

File: extensions/qxsck/var-and-list.js [hunk 1]
```diff
@@ -5,292 +5,872 @@
...
-                type: Scratch.ArgumentType.STRING,
+                type: "string",
                 defaultValue: "value",
               },
             },
-            extensions: ["colours_data"],
+          },
+          {
+            opcode: "seriVarsToJson",
+            blockType: "reporter",
```
Comment (LilyMakesThings) [inline extensions/qxsck/var-and-list.js:123]: Please use `Scratch.BlockType`. That's not a style thing, it's a compatibility thing.

File: extensions/qxsck/var-and-list.js [hunk 2]
```diff
@@ -5,292 +5,872 @@
...
+            text: Scratch.translate(
+              "convert all variables starting with [START] to json"
+            ),
+            arguments: {
+              START: {
+                type: "string",
```
Comment (LilyMakesThings) [inline extensions/qxsck/var-and-list.js:129]: Please use `Scratch.ArgumentType`. That's not a style thing, it's a compatibility thing.

File: extensions/qxsck/var-and-list.js [hunk 3]
```diff
@@ -5,292 +5,872 @@
...
+            {
+              text: Scratch.translate("delete the excess parts"),
+              value: "d",
+            },
+          ],
+        },
       };
     }
 
+    //from turbowarp-vm
```
Comment (LilyMakesThings) [inline extensions/qxsck/var-and-list.js:821]: no kidding.

File: extensions/qxsck/var-and-list.js [hunk 4]
```diff
@@ -5,292 +5,872 @@
...
+              LEFT: {
+                type: "string",
+                defaultValue: "1",
+              },
+              RIGHT: {
+                type: "string",
+                defaultValue: "5",
+              },
             },
-            extensions: ["colours_data_lists"],
```
Comment (LilyMakesThings) [inline extensions/qxsck/var-and-list.js:279]: These colours make the blocks the same colour as their intended categories. It's good for the recognition of blocks, so don't remove them.

File: extensions/qxsck/var-and-list.js [hunk 5]
```diff
@@ -5,292 +5,872 @@
...
+                type: "string",
+                menu: "associateList.List",
+              },
+            },
+          },
+
+          {
+            opcode: "unSupportCompile",
+            blockType: "label",
+            text: Scratch.translate("⚠️Compilation not supported"),
```
Comment (LilyMakesThings) [inline extensions/qxsck/var-and-list.js:716]: Compilation **is** supported, you're on TurboWarp, not Gandi IDE.

File: extensions/qxsck/var-and-list.js [hunk 6]
```diff
@@ -5,292 +5,872 @@
...
               },
+              DO: {
+                type: "string",
+                menu: "associateList.List",
+              },
+            },
+          },
+
+          {
+            opcode: "unSupportCompile",
```
Comment (LilyMakesThings) [inline extensions/qxsck/var-and-list.js:714]: You don't need an opcode for labels.

File: extensions/qxsck/var-and-list.js [hunk 7]
```diff
@@ -5,292 +5,872 @@
...
+            blockType: "label",
+            text: Scratch.translate("⚠️Compilation not supported"),
+            arguments: {},
```
Comment (LilyMakesThings) [inline extensions/qxsck/var-and-list.js:717]: You don't need arguments for labels.

File: extensions/qxsck/var-and-list.js [hunk 8]
```diff
@@ -5,292 +5,872 @@
...
+              RIGHT: {
+                type: "string",
+                defaultValue: "5",
+              },
             },
-            extensions: ["colours_data_lists"],
           },
         ],
+        menus: {
+          "openCaseSensitive.List": [
```
Comment (LilyMakesThings) [inline extensions/qxsck/var-and-list.js:768]: Menus should NOT be formatted like this: this locks them into not accepting reporters. See the [documentation on menus](https://docs.turbowarp.org/development/extensions/inputs#static-menus) for more details.

File: extensions/qxsck/var-and-list.js [hunk 9]
```diff
@@ -359,134 +1016,465 @@
...
+          }
         }
       }
       return 0;
     }
     getIndexesOfList(args, util) {
       /** @type {VM.ListVariable} */
       const variable = util.target.lookupVariableByNameAndType(
-        Scratch.Cast.toString(args.LIST),
+        String(args.LIST),
```
Comment (LilyMakesThings) [inline extensions/qxsck/var-and-list.js:1226]: Use Scratch.Cast.toString. Don't use "String".

**Review by LilyMakesThings — CHANGES_REQUESTED** [review]
I can tell this is just a pseudo-port of the Gandi IDE version of this extension. Please understand when you submit extensions to TurboWarp, we expect you to follow the standards and practises guided in the TurboWarp documentation and contributing guidelines. Things like `Scratch.Cast`, `Scratch.ArgumentType` and so on are generally not optional, especially if they were already used as removing those could break compatibility, especially in the future. If Gandi IDE decided not to support what was already in this extension prior, like `Scratch.Cast`, that is their problem, the same way we don't have hat parameters.

Furthmore, I don't really understand the point of this extension; I can see why it would be useful in Gandi IDE, a place that does not have access to the TurboWarp Extension Gallery, but over here a lot of these needs have already been filled.

**Comment by GarboMuffin** [comment]
haven't even read the above comments but this is really hard to review because the diff is the entire extension, which is rather unusual when updating something that was already working...

**Comment by LilyMakesThings** [comment]
> haven't even read the above comments but this is really hard to review because the diff is the entire extension, which is rather unusual when updating something that was already working...

tldr: it's a port of the gandi ide version, which has less APIs, so it was downgraded.

File: extensions/qxsck/var-and-list.js (hunk 4 above)
Comment (qxsck) [reply extensions/qxsck/var-and-list.js:279]: does now Turbowarp supports compiling C-type blocks?

File: extensions/qxsck/var-and-list.js (hunk 5 above)
Comment (qxsck) [reply extensions/qxsck/var-and-list.js:716]: does now Turbowarp supports compiling C-type blocks?

File: extensions/qxsck/var-and-list.js (hunk 8 above)
Comment (qxsck) [reply extensions/qxsck/var-and-list.js:768]: chang it now compromise compatibility

File: extensions/qxsck/var-and-list.js (hunk 1 above)
Comment (qxsck) [reply extensions/qxsck/var-and-list.js:123]: OK

File: extensions/qxsck/var-and-list.js (hunk 2 above)
Comment (qxsck) [reply extensions/qxsck/var-and-list.js:129]: OK

File: extensions/qxsck/var-and-list.js (hunk 3 above)
Comment (qxsck) [reply extensions/qxsck/var-and-list.js:821]: i read it in turbowarp-vm

File: extensions/qxsck/var-and-list.js (hunk 9 above)
Comment (qxsck) [reply extensions/qxsck/var-and-list.js:1226]: thank

File: extensions/qxsck/var-and-list.js (hunk 4 above)
Comment (qxsck) [reply extensions/qxsck/var-and-list.js:279]: different colors make it easier to distinguish them from the Scratch version

File: extensions/qxsck/var-and-list.js (hunk 5 above)
Comment (LilyMakesThings) [reply extensions/qxsck/var-and-list.js:716]: yes

Comment (qxsck) [reply extensions/qxsck/var-and-list.js]: > yes

ok,delete it just now
